components/org.wso2.carbon.identity.fraud.detection.sift/src/test/java/org/wso2/carbon/identity/fraud/detection/sift/conditional/auth/functions/PublishLoginToSiftFunctionImplTest.java

```diff
@@ -0,0 +1,153 @@
+    private HttpEntity httpEntity;
+
+    @InjectMocks
+    private PublishLoginToSiftFunctionImpl publishLoginToSiftFunction;
+
+    private MockedStatic<Util> utilMockedStatic;
+
+    private ByteArrayOutputStream logOutput;
+
+    @BeforeClass
+    public void setUp() throws FrameworkException {
+
+        MockitoAnnotations.openMocks(this);
+        utilMockedStatic = mockStatic(Util.class);
+        when(Util.getPassedCustomParams(any())).thenReturn(new HashMap<>());
+        when(Util.isLoggingEnabled(any())).thenReturn(true);
+        when(Util.buildPayload(any(), anyString(), anyMap())).thenReturn(new JSONObject());
+    }
+
+    @BeforeMethod
+    public void redirectOutputStreams() {
+
+        // Redirect System.out to capture logs.
+        logOutput = new ByteArrayOutputStream();
+        System.setOut(new PrintStream(logOutput));
+    }
+
+    @AfterClass
+    public void tearDown() {
+
+        utilMockedStatic.close();
+    }
+
+    @Test(priority = 0)
+    public void testPublishLoginEventToSift_Success() throws Exception {
+
+        when(httpClient.execute(any(HttpPost.class))).thenReturn(httpResponse);
+        StatusLine statusLine = mock(StatusLine.class);
+        when(httpResponse.getStatusLine()).thenReturn(statusLine);
+        when(httpResponse.getStatusLine().getStatusCode()).thenReturn(HttpStatus.SC_OK);
+        when(httpResponse.getEntity()).thenReturn(httpEntity);
+        when(httpEntity.getContent())
+                .thenReturn(new ByteArrayInputStream("{\"status\":0}".getBytes(StandardCharsets.UTF_8)));
+
+        publishLoginToSiftFunction.publishLoginEventToSift(
+                mock(JsAuthenticationContext.class), "LOGIN_SUCCESS", new ArrayList<>(),
+                new HashMap<String, Object>());
+
+        Assert.assertTrue(logOutput.toString().contains("Successfully published login event information to Sift."));
+    }
+
+    @Test(priority = 0)
+    public void testPublishLoginEventToSift_SiftError() throws Exception {
+
+        when(httpClient.execute(any(HttpPost.class))).thenReturn(httpResponse);
+        StatusLine statusLine = mock(StatusLine.class);
+        when(httpResponse.getStatusLine()).thenReturn(statusLine);
+        when(httpResponse.getStatusLine().getStatusCode()).thenReturn(HttpStatus.SC_OK);
+        when(httpResponse.getEntity()).thenReturn(httpEntity);
+        when(httpEntity.getContent())
+                .thenReturn(new ByteArrayInputStream("{\"status\":1}".getBytes(StandardCharsets.UTF_8)));
+
+        publishLoginToSiftFunction.publishLoginEventToSift(
+                mock(JsAuthenticationContext.class), "LOGIN_SUCCESS", new ArrayList<>(),
+                new HashMap<String, Object>());
+
+        Assert.assertTrue(logOutput.toString().contains("Error occurred from Sift while publishing login event" +
+                " information. Received Sift status: 1"));
+    }
+
+    @Test(priority = 1)
+    public void testPublishLoginEventToSift_ErrorResponse() throws Exception {
```

Comment:
Lets maintain the naming convention. Fix other places in the PR as well.
```suggestion
    public void testPublishLoginEventToSiftErrorResponse() throws Exception {
```